stackslib/src/net/tests/mod.rs

```diff
@@ -588,6 +641,29 @@ impl NakamotoBootPlan {
                             i += 1;
 
                             let mut txs = vec![];
+                            // check if the stacker/signers need to vote for an aggregate key. if so, append those transactions
+                            //  to the end of the block.
+                            // NOTE: this will only work the block after .signers is updated, because `make_all_signers_vote...`
+                            //  checks the chainstate as of `tip` to obtain the signer vector. this means that some tests may
+                            //  need to produce an extra block in a tenure in order to get the signer votes in place.
+                            //  The alternative to doing this would be to either manually build the signer vector or to refactor
+                            //  the testpeer such that a callback is provided during the actual mining of the block with a
+                            //  `ClarityBlockConnection`.
```

Comment:
This is good.  I'd rather the test framework faithfully simulate what signers ought to be doing (i.e. sending transactions) instead of directly filling data into a block.